lib/node/runner/checker.go

```diff
@@ -376,6 +380,35 @@ func BallotIsSameProposer(c common.Checker, args ...interface{}) (err error) {
 	return
 }
 
+// BallotRenewal checks that the ballot confirmed and proposed time is valid and
+// if the time is invalid, then it broadcasts expired ballot
+func BallotRenewal(c common.Checker, args ...interface{}) (err error) {
+	checker := c.(*BallotChecker)
+	nr := checker.NodeRunner
+	is := nr.Consensus()
+	blt := checker.Ballot
+
+	if !blt.State().IsValidForVote() {
+		return
+	}
+
+	needRenewal := is.RemoveOldBallots(blt)
+	if needRenewal {
+		expiredBlt := is.GenerateExpiredBallot(blt.VotingBasis(), blt.State())
+		nr.SetSendRecord(
+			consensus.ISAACState{
+				Height:      blt.VotingBasis().Height,
+				Round:       blt.VotingBasis().Round,
+				BallotState: blt.State(),
+			},
+			false,
```

Comment:
In this part, `nr.SetSendRecord` should be `true`? The expired Ballot is sent, right?

Reply:
@spikeekips Oh, good question 👍 
This false setting is for initializing SendRecord.
Because if SendRecord is true, then this ballot will be discarded in nr.BroadcastBallot().
If it is confusing, I can make this to another method like `InitializeSendRecord`.

Reply:
@Charleslee522 Renaming will be good :)